src/dataset/infrastructure/repositories/DatasetJSDataverseRepository.ts

```diff
@@ -147,10 +148,14 @@ export class DatasetJSDataverseRepository implements DatasetRepository {
       })
       .catch((error: ReadError) => {
         console.error(error)
-        if (!version) {
+        if (version === DatasetNonNumericVersion.LATEST_PUBLISHED) {
           throw new Error(`Failed to get dataset by persistent ID: ${error.message}`)
         }
```

Comment:
question: why only throwing the error if the requested version was :latest-published ?
I understand the need of refetching and saving the original requested version but what happen if there is an error when trying to fetch a :draft or :latest version?

Reply:
The assumption with this logic is that the error was thrown because the version doesn't exist.  We don't currently have a specific error type to catch when the version isn't found, but there are plans to add one: https://github.com/IQSS/dataverse-frontend/issues/363 .  You're right that we should have separate logic for handling other types of errors, but I thought it was better to keep it like this for now, rather than adding logic that will be replaced later.  So what will happen for now, is that the UI will show 'page not found' if both the requested version and the :latest-published version throw an error, for whatever reason.  Let me know what you think, if that's ok, or we should add more logic for handling different types of errors.

Reply:
You are right, I think it is fine to leave it as it is for now, as you mention this will probably change with the 363 integration.